Adding battles to the game

milkyroute opened the pull request:
I think the battles' foundations are now ready. There is no animations, but the damages are working and the health bars too.
If you want me to do some things before merging, say it below.

milkyroute committed:
Add battle fondations

milkyroute committed:
Add two OpMon battle initialization

milkyroute committed:
Add turn order selection

milkyroute committed:
Little modifications

milkyroute committed:
Add attack possibility verification

milkyroute committed:
Rechecked Attack::attack

milkyroute committed:
Adds strings in turn and keys files
Improved a little OpString

milkyroute committed:
Add attacks effects in Turn

milkyroute committed:
Moved Turn in Model.
It now has it's own file !

milkyroute committed:
Fix some compilation errors

milkyroute committed:
Commented some code for the contributors

milkyroute committed:
Fixed some compilation errors
TODO : Fix the OpString problem :
The constructor have changed and I try to make it easier. The current problem is that I want to be able to put as much
parameters as I want, but without va_args. I must find a solution about this.

milkyroute committed:
Return of va_args
Sorry ^^ I think it is more handy. However, if you prefer, you can still use the constructor with a vector.

milkyroute committed:
Fixed compilation errors, still link errors

milkyroute committed:
Fixed compilation errors for the battle

milkyroute committed:
Added Battle's constructor and some variables

milkyroute committed:
Adds resources variables in Data.[c|h]pp

milkyroute committed:
Adds resources' loading in Initializer
+ Edited some Jlppc sprites, and renamed them "cyrion"

milkyroute committed:
Fixed segfault

navet56 committed:
Add battle_dialog.png (the Dialog box for the battles)

navet56 committed:
Big work about the OpMons
Renaming
New OPMons : Prestazuli and Sautinelle
New OpMons back sprites : Chibiro

navet56 committed:
Add many back sprites of OpMons

navet56 committed:
Add a lot of back sprites of OPMons

milkyroute committed:
Adding the first fight preview

milkyroute committed:
Fixed compilation errors

milkyroute committed:
Battle screen can now be printed on the screen
Still, the positionment and objects' size are incorrect in some cases,
but it works!
You can access to the battle screen by pressing 'B' while in debug mode
anywhere in the overworld.

milkyroute committed:
Fixed positions

navet56 committed:
Update OpMons stats in the code

navet56 committed:
Fix a compilation bug

navet56 committed:
Update positions

milkyroute committed:
Merge master commits into battle

navet56 committed:
Fix a bug

milkyroute committed:
Fixed compilation errors

navet56 committed:
Rename dialog sprites

navet56 committed:
Add the wild battle music in Data

navet56 committed:
Update trainer and player battle's sprites

milkyroute committed:
The cursor now moves

milkyroute committed:
The OpMons are now visible

milkyroute committed:
The opmons' data are now visible
The name, the level and the health (The health bar is not functional)

navet56 committed:
IMprovement of the battle interface

navet56 committed:
Fix choice text font size

navet56 committed:
Update choice interface

milkyroute committed:
Fixed battle cursor's positions

navet56 committed:
Battle interface improvement

navet56 committed:
Improve battle interface

navet56 committed:
Update battle_dialog.png

navet56 committed:
Add a new square for the Info Box of the Trainer

navet56 committed:
Add shadows in the battle screen

navet56 committed:
Fix OpMons stats

BAKFR committed:
Use AttackEffect classes as strategy of Attack class

BAKFR committed:
Make global `camUnlock` an attribute of Overworld view

BAKFR committed:
Let some space inside the camera to move the player.

navet56 committed:
Add the attack box for the battles

milkyroute committed:
Changed camera coef

navet56 committed:
Update atk box

navet56 committed:
Added Silvermoon and Lebordelic to credits (I forgot)

BAKFR committed:
StartScene: trim the fat

BAKFR committed:
dialog: git rid of static members; Improve interface

BAKFR committed:
Overworld: print UI elements with default sf::View instead of using PixelToCoords

BAKFR committed:
Overword: Better camera focus on reset near the map borders, and for small maps

BAKFR committed:
Overworld: fix regression on main character animation

navet56 committed:
Addition of 2 new OpMons's sprites

navet56 committed:
Update the OpMon 37

milkyroute committed:
Add text in the second battle dialog
OpString : add quickString function, allowing to replace a string
quickly with std::string pointers instead of sf::String ones. Returns
directly the final string, and deletes the pointers directly.

milkyroute committed:
Modified the attacks choice's dialog's texture

navet56 committed:
Add the sprites of OpMon's types

navet56 committed:
Rkeys : Update battle traduction

milkyroute committed:
OpMon : Fixed an inappropriate name
OpMon::falsif -> OpMon::initialized

navet56 committed:
Updated Oursiflamme back sprite

navet56 committed:
Merge branch 'battle' of https://github.com/cyrioncentori/OpMon into battle

BAKFR committed:
Camera: prevent the camera to display too much out-of-map space

BAKFR committed:
Add indoor Map attribute; Camera never display "void" on outdoor maps

BAKFR committed:
Remove the 4 first columns of Road14

navet56 committed:
Add new back sprites of OpMons
OPMon 08 and 09

milkyroute committed:
Add attack selection interface
(Just press return when the cursor are below "Attack")

milkyroute committed:
Attacks' type is now visible
on the attack selection interface

milkyroute committed:
Press backspace to return to the action selection interface
from the attack selection interface

navet56 committed:
Added the charge attack animation (alpha test)

navet56 committed:
Updated credits screen

navet56 committed:
Update Chimney smoke animation (finally...)

navet56 committed:
Slowing the smoke speed

milkyroute committed:
Fix #46 (Just an incorrect map id)

navet56 committed:
OpCenter: Fix bad TP

milkyroute committed:
Attacks now work! (No animation)
Heath bars work too, and when an OpMon is K.O, the game returns to the
overworld.

navet56 committed:
Updated Resources folder

navet56 committed:
Updated Start Scene Window animation

milkyroute committed:
Fixed crash and attacks with 0 PP are unsusable
Now, nothing is done if an invalid attack is used

milkyroute committed:
Pressing run now closes the battle

navet56 reviewed (changes requested):
Il faut rajouter les textes qui disent quelle attaque le joueur et le dresseur font, la on ne comprend pas

navet56 committed:
Add Asvirius to the credits

navet56 committed:
Merge branch 'battle' of https://github.com/cyrioncentori/OpMon into battle

navet56 committed:
Fixed #49

navet56 committed:
Fix typo in the Debug Mode

navet56 committed:
Debug mode : Fix typo (2)

navet56 committed:
Updated a lot of musics

milkyroute committed:
Fixed a windows compilation error

milkyroute committed:
Preparing dialogs during a turn
I'm talking about the dialogs like "OpMon is using Attack!" or something like that.

navet56 committed:
Added 3 new OpMon's back sprites

navet56 committed:
Battle Rkeys : Added attack dialog

milkyroute committed:
Adding mid-battle dialogs (1/?)

milkyroute committed:
Adding mid-battle dialogs (2/?)

milkyroute committed:
Adding mid-battle dialogs (3/?)
/!\ Segfault, battles do not work anymore. If you are searching
for a working battle, go to this commit : cb4d0f6

navet56 committed:
Add opmon choice background

navet56 committed:
Update mom sprite
+ Update battle interface

milkyroute committed:
Deleted a useless file

milkyroute committed:
Adding mid-battle dialogs (4/?)
Fixed segfault

milkyroute committed:
Adding mid-battle dialogs (5/?)
Now functional, still some problems

milkyroute committed:
Adding mid-battle dialogs (6/?)
Pressing Return now passes to the next dialog (Like space)

milkyroute committed:
Adding mid-battle dialogs (7/7)
Fixed HP actualization (It was in the wrong order)
